sub/source/docs/modes.rst

```diff
@@ -0,0 +1,62 @@
+.. _modes:
+
+Sub Modes
+=========
+Sub has the following modes:
+
+legend for requirements: P - needs Position(GPS,etc.), D - needs Depth sensor
+
+=============   =================================================  =========
+Mode            Description                                        Requires
+=============   =================================================  =========
+MANUAL          Pilot control with no stabilization                 -
+ACRO            Body-frame rate control, manual depth control       -
+STABILIZE       Manual angle control,manual depth control           -
+ALT_HOLD        Stabilize with automatic depth control              D
+AUTO            Automatic mission command list execution            P/D
+GUIDED          Swim to location or velocity/direction using GCS    P/D
+CIRCLE          Circle swim with depth control                      P/D
+SURFACE         Return to surface, pilot directional control        P/D
+POSHOLD         Loiter with depth control and pilot overrides       P/D
+MOTOR_DETECT    Automatically determine motor rotation and adjust   D
+SURFTRAK        Hold distance above seafloor while stabilizing      -
+=============   =================================================  =========
+
+Mode Specific Parameters
+========================
+
+CIRCLE Mode
+-----------
+* :ref:`CIRCLE_RADIUS<CIRCLE_RADIUS>`
+* :ref:`CIRCLE_RATE<CIRCLE_RATE>`
+* :ref:`CIRCLE_OPTIONS<CIRCLE_OPTIONS>`
+
+Depth Control Related
+---------------------
+* :ref:`SURFACE_DEPTH<SURFACE_DEPTH>`
+* :ref:`PILOT_SPEED_UP<PILOT_SPEED_UP>`
+* :ref:`PILOT_SPEED_DN<PILOT_SPEED_DN>`
+* :ref:`PILOT_SPEED<PILOT_SPEED>`
+
+Stabilization Related
+---------------------
+* :ref:`ANGLE_MAX<ANGLE_MAX>`
```

Comment:
`JS_GAIN` also reduces the maximum angle when it's being manually set by a joystick, because it scales the joystick inputs.

Reply:
added